### PR TITLE
Release on GitHub

### PR DESCRIPTION
**Pull request by rustamwin**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | #10

### CODE REVIEW

File: src/App/Command/Release/MakeCommand.php [hunk 1]
```diff
@@ -214,4 +221,40 @@ private function getVersionToRelease(Version $currentVersion): Version
...
+
+    private function releaseOnGithub(Package $package, Version $versionToRelease): void
+    {
+        $io = $this->getIO();
+        $io->info("Creating release on GitHub for $versionToRelease.");
+
+        $client = new Client();
+        try {
+            $token = $this->getToken();
+        } catch (\Exception $e) {
```
Comment (arogachev) [inline src/App/Command/Release/MakeCommand.php:233]: Is it possible to catch a more specific exception here?

File: src/App/Command/Release/MakeCommand.php [hunk 2]
```diff
@@ -214,4 +221,40 @@ private function getVersionToRelease(Version $currentVersion): Version
...
+        $changelogPath = $package->getPath() . '/CHANGELOG.md';
+        $changelog = new Changelog($changelogPath);
+        $client->authenticate($token, null, AuthMethod::ACCESS_TOKEN);
+        $release = new Releases($client);
+        $release->create($package->getVendor(), $package->getId(), [
+            'name' => sprintf('Release %s (%s)', $versionToRelease, date('F d, Y')),
+            'tag_name' => $versionToRelease->asString(),
+            'body' => implode("\n", $changelog->getReleaseNotes($versionToRelease)),
+            'draft' => false,
+            'prerelease' => false // TODO: check if this is pre-release
```
Comment (arogachev) [inline src/App/Command/Release/MakeCommand.php:247]: Suggest to create an issue instead of `TODO`.

File: src/Infrastructure/Changelog.php [hunk 3]
```diff
@@ -65,6 +65,13 @@ public function close(Version $version): void
         );
     }
 
+    public function getReleaseNotes(Version $version): array
```
Comment (arogachev) [inline src/Infrastructure/Changelog.php:68]: Add PHPDoc tag to specify which elements array contains (for example, `string[]`, etc.).

**Review by arogachev — CHANGES_REQUESTED** [review]

**Review by arogachev — APPROVED** [review]

File: src/App/Command/Release/MakeCommand.php [hunk 4]
```diff
@@ -169,6 +172,10 @@ protected function processPackage(Package $package): void
             $git->push();
             $git->pushTags();
 
+            if ($this->confirm('Release on GitHub?')) {
```
Comment (vjik) [inline src/App/Command/Release/MakeCommand.php:175]: Maybe modify previous question to "Push commits and tags, and release on GitHub?" and remove this confirmation?

File: src/Infrastructure/Changelog.php [hunk 5]
```diff
@@ -65,6 +65,18 @@ public function close(Version $version): void
...
     }
 
+    /**
+     * @param Version $version
+     *
+     * @return string[]
+     */
+    public function getReleaseNotes(Version $version): array
+    {
+        [, $changelog,] = $this->splitChangelog($version);
```
Comment (vjik) [inline src/Infrastructure/Changelog.php:75]: Suggest remove any empty lines.

File: src/App/Command/Release/MakeCommand.php [hunk 6]
```diff
@@ -214,4 +221,40 @@ private function getVersionToRelease(Version $currentVersion): Version
...
+            'tag_name' => $versionToRelease->asString(),
+            'body' => implode("\n", $changelog->getReleaseNotes($versionToRelease)),
+            'draft' => false,
+            'prerelease' => false, // TODO: check if this is pre-release
+        ]);
+    }
+
+    private function getToken(): string
+    {
+        $tokenFile = $this->getAppRootDir() . 'config/github.token';
```
Comment (vjik) [inline src/App/Command/Release/MakeCommand.php:253]: Add info about token to [readme](https://github.com/yiisoft/yii-dev-tool#configuration) in "Configuration" section.

Comment (rustamwin) [reply src/App/Command/Release/MakeCommand.php:251]: Added.

File: src/App/Command/Release/MakeCommand.php (hunk 4 above)
Comment (rustamwin) [reply src/App/Command/Release/MakeCommand.php]: Good idea.

File: README.md [hunk 7]
```diff
@@ -71,6 +71,9 @@ To customize the configuration of **yii-dev-tool**, create your local configurat
 using file `packages.local.php.example` as example. In this file you will find examples of all available 
 configuration options.
 
+Some commands such as `github/protect/branch`, `github/settings`, `release/make` require a Github auth token. Therefore, you need to create a personal access token on Github.
```
Comment (vjik) [inline README.md:74]:
```suggestion
Some commands such as `github/protect/branch`, `github/settings`, `release/make` require a GitHub auth token. Therefore, you need to create a personal access token on GitHub.
```

**Pull request merged**